src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDaoTest.java

```diff
@@ -669,12 +672,216 @@ public void constraintRulesOnQuestionOverwriteConstraintRules() {
         when(mockElementResultsPage.getResults()).thenReturn(ImmutableList.of((DynamoSurveyElement)question));
         
-        Survey result = dao.getSurvey(SURVEY_KEYS, true);
+        Survey result = dao.getSurvey(TEST_STUDY, SURVEY_KEYS, true);
         SurveyQuestion resultQuestion = result.getUnmodifiableQuestionList().get(0);
         SurveyRule rule = resultQuestion.getConstraints().getRules().get(0);
         assertTrue(rule.getEndSurvey());
     }
     
+    @Test
+    public void getSurveyIncludeElementsByIdentifer() {
+        DynamoSurvey survey = new DynamoSurvey();
+        survey.setStudyIdentifier(TEST_STUDY_IDENTIFIER);
+        survey.setGuid(GUID);
+        survey.setIdentifier(SURVEY_ID);
+        survey.setCreatedOn(TIMESTAMP);
+        mockSurveyMapper(survey);
+        
+        // There is a question that should be retrieved.
+        DynamoSurveyQuestion element = new DynamoSurveyQuestion();
+        element.setType(SurveyElementConstants.SURVEY_QUESTION_TYPE);
+        element.setUiHint(UIHint.BLOODPRESSURE);
+        element.setConstraints(new BloodPressureConstraints());
+        when(mockSurveyElementMapper.queryPage(eq(DynamoSurveyElement.class), any())).thenReturn(mockElementResultsPage);
+        when(mockElementResultsPage.getResults()).thenReturn(ImmutableList.of(element));
+        
+        Survey result = dao.getSurvey(TEST_STUDY, SURVEY_IDENTIFIER_KEYS, true);
+        assertSame(result, survey);
+        assertEquals(result.getElements().size(), 1);
+        
+        verify(mockSurveyMapper).queryPage(eq(DynamoSurvey.class), queryCaptor.capture());
+        DynamoDBQueryExpression<DynamoSurvey> query = queryCaptor.getValue();
+        assertEquals(query.getHashKeyValues().getStudyIdentifier(), TEST_STUDY_IDENTIFIER);
+        
+        verify(mockSurveyElementMapper).queryPage(eq(DynamoSurveyElement.class), elementQueryCaptor.capture());
+        DynamoDBQueryExpression<DynamoSurveyElement> elementQuery = elementQueryCaptor.getValue();
+        assertEquals(elementQuery.getHashKeyValues().getSurveyCompoundKey(), GUID + ":" + TIMESTAMP);
+    }
+    
+    @Test
+    public void getSurveyExcludeElementsByIdentifier() {
+        DynamoSurvey survey = new DynamoSurvey();
+        survey.setGuid(GUID);
+        survey.setStudyIdentifier(TEST_STUDY_IDENTIFIER);
+        survey.setIdentifier(SURVEY_ID);
+        survey.setCreatedOn(TIMESTAMP);
+        mockSurveyMapper(survey);
+        
+        // There is a question that should be retrieved.
+        DynamoSurveyQuestion element = new DynamoSurveyQuestion();
+        element.setType(SurveyElementConstants.SURVEY_QUESTION_TYPE);
+        element.setUiHint(UIHint.BLOODPRESSURE);
+        element.setConstraints(new BloodPressureConstraints());
+        when(mockSurveyElementMapper.queryPage(eq(DynamoSurveyElement.class), any())).thenReturn(mockElementResultsPage);
+        when(mockElementResultsPage.getResults()).thenReturn(ImmutableList.of(element));
```

Comment:
There is no need to mock the mockSurveyElementMapper.queryPage(), as it will never be called.